modules/calib3d/src/usac.hpp

```diff
@@ -0,0 +1,549 @@
+     * @norm_points is output matrix of size pts_size x 4
+     * @sample constains indices of points
+     * @sample_number is number of used points in sample <0; sample_number)
+     * @T1, T2 are output transformation matrices
+     */
+    virtual void getNormTransformation (Mat &norm_points, const std::vector<int> &sample,
+                                        int sample_number, Mat &T1, Mat &T2) const = 0;
+    static Ptr<NormTransform> create (const Mat &points);
+};
+
+/////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////// SOLVER ///////////////////////////////////////////
+class MinimalSolver : public Algorithm {
+public:
+    // Estimate models from minimal sample. models.size() == number of found solutions
+    virtual int estimate (const std::vector<int> &sample, std::vector<Mat> &models) const = 0;
+    // return minimal sample size required for estimation.
+    virtual int getSampleSize() const = 0;
+    // return maximum number of possible solutions.
+    virtual int getMaxNumberOfSolutions () const = 0;
+    virtual Ptr<MinimalSolver> clone () const = 0;
+};
+
+//-------------------------- HOMOGRAPHY MATRIX -----------------------
+class HomographyMinimalSolver4ptsGEM : public MinimalSolver {
+public:
+    static Ptr<HomographyMinimalSolver4ptsGEM> create(const Mat &points_);
+};
+
+//////////////////////////////////////// NON MINIMAL SOLVER ///////////////////////////////////////
+class NonMinimalSolver : public Algorithm {
+public:
+    // Estimate models from non minimal sample. models.size() == number of found solutions
+    virtual int estimate (const std::vector<int> &sample, int sample_size,
+          std::vector<Mat> &models, const std::vector<double> &weights) const = 0;
+    // return minimal sample size required for non-minimal estimation.
+    virtual int getMinimumRequiredSampleSize() const = 0;
+    // return maximum number of possible solutions.
+    virtual int getMaxNumberOfSolutions () const = 0;
+    virtual Ptr<NonMinimalSolver> clone () const = 0;
+};
+
+//-------------------------- HOMOGRAPHY MATRIX -----------------------
+class HomographyNonMinimalSolver : public NonMinimalSolver {
+public:
+    static Ptr<HomographyNonMinimalSolver> create(const Mat &points_);
+};
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////// SCORE ///////////////////////////////////////////
+class Score {
+public:
+    int inlier_number;
+    double score;
+    Score () { // set worst case
+        inlier_number = 0;
+        score = std::numeric_limits<double>::max();
+    }
+    Score (int inlier_number_, double score_) { // copy constructor
+        inlier_number = inlier_number_;
+        score = score_;
+    }
+    // Compare two scores. Objective is minimization of score. Lower score is better.
+    inline bool better(const Score &score2) const {
```

Comment:
`bool isBetter()`, like `isInlier()` below.

Reply:
@ivashmak, please, rename the method for better consistency

Reply:
fixed